src/qibolab/dummy.py

```diff
@@ -100,15 +100,17 @@
 
 def create_dummy():
     """Create a dummy platform using the dummy instrument."""
+    # Create dummy controller
+    instrument = DummyInstrument(NAME, 0)
 
     # Create channel objects
     channels = ChannelMap()
-    channels |= "readout"
-    channels |= (f"drive-{i}" for i in range(RUNCARD["nqubits"]))
-    channels |= (f"flux-{i}" for i in range(RUNCARD["nqubits"]))
+    channels["readout"] = Channel("readout", port=instrument["readout"])
```

Comment:
A couple of possible improvements:

1. in this layout the channel name is repeated: it could be avoided making the name optional in `Channel`, and setting it through `ChannelMap.__setitem__`, essentially adding `channel.name = name`
2. since the name is actually repeated three times, there is room for further improvement: essentially, the information to be usually passed to `ChannelMap` is the instrument, that is frequently the same for a batch of channels (if not all), a couple of options are:

```py
# augmenting `from_names` with the current instrument

class ChannelMap:
    @classmethod
    def from_names(cls, *names):
        return cls({name: Channel(name, port=cls.instrument[name]) for name in names})


channels.instrument = instrument
channels |= "readout"
channels |= (f"drive-{i}" for i in range(RUNCARD["nqubits"]))

# adding a new constructor
class ChannelMap:
    @classmethod
    def from_pairs(cls, *pairs):
        return cls({name: Channel(name, instr[name]) for name, instr in pairs})

channels |= ChannelMap.from_pairs([("readout", instrument)])
channels |= ChannelMap.from_pairs((f"drive-{i}", instrument) for i in range(RUNCARD["nqubits"]))
```

I actually prefer the first option, since it maintains the present conciseness. But nothing prevents to implement both (apart from the *Zen of Python*, who suggests to only have a single way of doing things...).

Reply:
Thank you for these suggestions. Please have a look at the last commit https://github.com/qiboteam/qibolab/pull/469/commits/004ae377ed7238a69bfcf82a99249f256164b2b7 where I attempted to simplify this, particularly remove the channel name repetition in platform `create` methods.

The idea is that `channels |= ...` can take `str`, `Channel` or an iterable containing any of these two types. If it takes a `Channel` it just adds it to the dictionary, while if it takes a `str` it creates an empty `Channel` (without ports) and adds it to the dictionary. 

This provides different ways to the user to define the channels. For example one can define channels and assign ports at the same time:
```py
channels = ChannelMap()
channels |= Channel("L1-1", port=instrument["o1"])
channels |= Channel("L1-2", port=instrument["o2"])
```
or define all channels and assign ports later, as we were doing before:
```py
channels = ChannelMap()
channels |= ("L1-1", "L1-2")

channels["L1-1"].port =instrument["o1"]
channels["L1-2"].port = instrument["o2"]
```
Both are supported.

Reply:
Regarding this discussion, see also #487.